apps/crn-server/src/data-providers/contentful/research-output.data-provider.ts

```diff
@@ -416,8 +418,8 @@ const prepareInput = (
     ),
     labs: getLinkEntities(labIds),
     teams: getLinkEntities(teamIds),
-    relatedResearch: getLinkEntities(relatedResearchIds),
-    relatedEvents: getLinkEntities(relatedEventIds),
+    relatedResearch: relatedResearchIds && getLinkEntities(relatedResearchIds),
+    relatedEvents: relatedEventIds && getLinkEntities(relatedEventIds),
```

Comment:
What's the reason for this change? There are several other calls to `getLinkEntities` here that _don't_ have the check, which seems odd. Surely it should be all of them or none of them?

Reply:
Fix Typescript error 
```
Argument of type 'string[] | undefined' is not assignable to parameter of type 'string[]'.
  Type 'undefined' is not assignable to type 'string[]'
```

Reply:
How come that only applies to `relatedResearch` and `relatedEvents` and not also the labs, teams, methods, keywords, etc...?

Reply:
So i reverted this change as it was a rebasing issue `relatedResearchIds` and `relatedEventIds` was optional when i first started working on this now they are required.